FluxNotes/flux
# Helpful message when missing parent context

Supports a help message when a shortcut is incomplete because a parent shortcut is missing. Additionally had to make some other changes in the form of refactoring and code additions.:
- Cleaned up the getHelp completion component a bit. 
- Pass shortcutManager to completionComponents now.
- Added labels to any shortcuts that show up as KnownParentContexts. These labels, in addition to being used as a placeholder for the shortcut when being expanded (often not a problem for any of the shortcuts touched here), are a form of readable text to present to the user. Thus we use shortcut.label when trying to figure out how to display something in the GetHelp portal. 
- CompletionPortalSearchIndices now use TriggersWithoutLabel to define their filtered list of shortcuts.
- Updated the TriggersWithoutLabel function to remove one instance of a label in thelist of triggers instead of all instances. This avoids issues with shortcuts like @condition, where the label is identical to one of its string triggers. 

To test basic functionality: 
- Insert #disease status into the editor, and open the getHelp menu. Confirm that the message displayed is sufficiently helpful. 

To test fully...
- I think we need to give a test on all the shortcuts we have. I tried to test this as much as I could but I'm worried I might have missed a few cases. 

_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._


## Submitter:

**Running the Application**


**Task**


**Documentation**


**Code Quality**


**Tests**



## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code
